feat(stark-build): add support to extend index-html.transform.js (#4056)
NationalBankBelgium/stark: docs/stark-build/NG_CLI_BUILD_CUSTOMIZATIONS.md

}
```

#### If you need to override `starkAppMetadata` or `starkAppConfig` depending on your target configuration you can extend `indexTransform`

```text
{
    ...
    "projects": {
        "your-app": {
            ...
            "architect": {
                "build": {
                    ...
                    "options": {
                        ...

>                         "indexTransform": ".config/index-html.transform.js"  // default value: "./node_modules/@nationalbankbelgium/stark-build/config/index-html.transform.js"

Reviewer: This should be

```suggestion
                        "indexTransform": "./config/index-html.transform.js"  // default value: "./node_modules/@nationalbankbelgium/stark-build/config/index-html.transform.js"
```

}
```

> #### If you need to override `starAppMetadata`  or `starkAppConfig` depending on your target you can extends `indexTransform`

Reviewer: Typos + add missing `configuration` word.

```suggestion
#### If you need to override `starkAppMetadata`  or `starkAppConfig` depending on your target configuration you can extend `indexTransform`
```

```text
{
    ...
    "projects": {
        "your-app": {
            ...
            "architect": {

>                 "yourTargetName": {

Reviewer: This should be "build". It the index-html.transform script, we can distinguish the target based on the chosen configuration (`--configuration production)`.

```suggestion
                "build": {
```

                    ...
                    "options": {
                        ...

>                         "indexTransform": "path/to/your/new/config.js"  // default value: "./node_modules/@nationalbankbelgium/stark-build/config/index-html.transform.js"

Reviewer: I'd recommend to adapt this as following.
As we already force to add the `./config/index-head-config.js`, it would make sense to suggest a specific path to the developer.

```suggestion
                        "indexTransform": "./config/index-html.transform.js"  // default value: "./node_modules/@nationalbankbelgium/stark-build/config/index-html.transform.js"
```

                    }
                }
            }
        }
    }
}
```

> in your new index-transform

Reviewer: Following the comment above, I'd recommend to keep the same name as in Stark since we override the original file:

```suggestion
In your new index-html.transform.js
```

```javascript
"use strict"

> const starkHtmlTransformIndex = require("../node_modules/@nationalbankbelgium/stark-build/config/index-html.transform");

Reviewer: Typo + question about the require, could we remove the relative path like this  `require("@nationalbankbelgium/stark-build/config/index-html.transform");` ?

```suggestion
const starkIndexHtmlTransform = require("@nationalbankbelgium/stark-build/config/index-html.transform");
```

> const overridedData = {

Reviewer: Typo

```suggestion
const overriddenData = {
```

	starkAppConfig : {
		// your app config
	},
	starkAppMetadata : {
		// your app metadata 
	}

> }

Reviewer: I would recommend to extend the Stark common-data and override the desired properties:

```suggestion
const overriddenData = {
    ...
	starkAppConfig : {
		// your app config
	},
	starkAppMetadata : {
		// your app metadata 
	}
}
```

module.exports = (targetOptions, indexHtml) => {
	if (targetOptions.configuration === 'config_name_where_override') {
		return starkHtmlTransformIndex(targetOptions, indexHtml, overridedData);
	}
	else {
		// continue to use default options
		return starkHtmlTransformIndex(targetOptions, indexHtml);

> 	}

Reviewer: Following the comment above:

```suggestion
module.exports = (targetOptions, indexHtml) => {
	if (targetOptions.configuration === 'config_name_where_override') {
		return starkIndexHtmlTransform(targetOptions, indexHtml, overriddenData);
	}
	else {
		// continue to use default options
		return starkIndexHtmlTransform(targetOptions, indexHtml);
	}
```
